tests/acceptance/features/coreApiSharePublicLink3/updatePublicLinkShare.feature

```diff
@@ -610,3 +610,28 @@ Feature: update a public link share
       | dav-path | ocs_api_version | ocs_status_code |
       | spaces   | 1               | 100             |
       | spaces   | 2               | 200             |
+
+
+  Scenario Outline: Updating the role of a public link to internal
+    Given using OCS API version "<ocs_api_version>"
+    And using <dav-path> DAV path
+    And user "Alice" has uploaded file "filesForUpload/textfile.txt" to "/textfile.txt"
+    And user "Alice" has created a public link share with settings
+      | path        | /textfile.txt |
+      | permissions | read          |
+    When user "Alice" updates the last public link share using the sharing API with
+      | permissions | 0 |
+    Then the OCS status code should be "<ocs_status_code>"
+    And the HTTP status code should be "200"
```

Comment:
This is good, to test that the API call can be done to set the permissions to 0 (which means that anyone who accesses the pubic link will have to authenticate as a real user (i.e. it becomes a "link share" that is not really "public")

Are we also going to add API scenarios somewhere that confirm that the updated "link share" really does now work properly - a "public" API request to get the link share contents should fail, a request by a real user should pass.

Reply:
positive scenario we check in the e2e test: https://github.com/owncloud/web/pull/8260